src/khoj/routers/api_config.py

```diff
@@ -278,7 +279,76 @@ async def update_search_model(
     return {"status": "ok"}
 
 
-# Create Routes
+@api_config.post("/data/phone", status_code=200)
+@requires(["authenticated"])
+async def update_phone_number(
+    request: Request,
+    phone_number: str,
+    client: Optional[str] = None,
+):
+    user = request.user.object
+
+    await adapters.aset_user_phone_number(user, phone_number)
+
+    if is_twilio_enabled():
+        create_otp(user)
+    else:
+        logger.warning("Phone verification is not enabled")
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="set_phone_number",
+        client=client,
+        metadata={"phone_number": phone_number},
+    )
+
+    return {"status": "ok"}
+
+
+@api_config.delete("/data/phone", status_code=200)
+@requires(["authenticated"])
+async def delete_phone_number(
+    request: Request,
+    client: Optional[str] = None,
+):
+    user = request.user.object
+
+    await adapters.aremove_phone_number(user)
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="delete_phone_number",
+        client=client,
+    )
+
+    return {"status": "ok"}
+
+
+@api_config.post("/data/phone/otp", status_code=200)
```

Comment:
Should we use `/phone/verify` as URL for phone verification? This may better show the API endpoints purpose. OTP in URL doesn't make it clear whether OTP is being sent or verified